tests/flot.highlights.Test.js

```diff
@@ -0,0 +1,148 @@
+                    highlightPoints: true,
+                    highlightBars: false,
+                    lineWidth: 5,
+                    selectedIndexes:selectedIndexes
+                }
+            }
+        });
+        var spy = spyOn(plot.drawSymbol, 'square').and.callThrough();
+        plot.triggerRedrawOverlay();
+        jasmine.clock().tick(20);
+        expect(spy).toHaveBeenCalled();
+    });
+    it('should highlight a range of points', function () {
+        selectedRange = [[0, 5], [], []];
+        plot = $.plot("#placeholder", [ d1, d2, d3 ], {
+            series: {
+                lines: { show: true },
+                points: { show: true, symbol: 'square' },
+                highlights: {
+                    show: true,
+                    highlightColor: '#00FF00',
+                    highlightLines: false,
+                    highlightPoints: true,
+                    highlightBars: false,
+                    lineWidth: 5,
+                    selectedRange:selectedRange
+                }
+            }
+        });
+        var spy = spyOn(plot.drawSymbol, 'square').and.callThrough();
+        plot.triggerRedrawOverlay();
+        jasmine.clock().tick(20);
+        expect(spy).toHaveBeenCalled();
+    });
+    it('should not highlight points when show is false', function () {
+        selectedIndexes = [[0], [], []];
+        plot = $.plot("#placeholder", [ d1, d2, d3 ], {
+            series: {
+                lines: { show: true },
+                points: { show: true, symbol: 'square' },
+                highlights: {
+                    show: false,
+                    highlightColor: '#00FF00',
+                    highlightLines: false,
+                    highlightPoints: true,
+                    highlightBars: false,
+                    lineWidth: 5,
+                    selectedIndexes:selectedIndexes
+                }
+            }
+        });
+        var spy = spyOn(plot.drawSymbol, 'square').and.callThrough();
+        plot.triggerRedrawOverlay();
+        jasmine.clock().tick(20);
+        expect(spy).not.toHaveBeenCalled();
+    });
+    it('should highlight lines', function () {
+        selectedIndexes = [[0], [], []];
+        plot = $.plot("#placeholder", [ d1, d2, d3 ], {
+            series: {
+                lines: { show: true },
+                points: { show: true },
+                highlights: {
+                    show: true,
+                    highlightColor: '#00FF00',
+                    highlightLines: true,
+                    highlightPoints: false,
+                    highlightBars: false,
+                    lineWidth: 5,
+                    selectedIndexes:selectedIndexes
+                }
+            }
+        });
+        var spy = spyOn($.plot.drawSeries, 'drawSeriesLines').and.callThrough();
+        plot.triggerRedrawOverlay();
+        jasmine.clock().tick(20);
+        expect(spy).toHaveBeenCalled();
```

Comment:
To be clear, this isn't testing that your highlighting code was called correct? I think this comment applies to all of the tests.

Reply:
After the triggerRedrawOverlay the drawSeriesLines should have been called by the highlights plugin. So this IS testing that the highlighting code is called. drawSeriesLines won't be called by redrawOverlay if nothing is in the overlay.